File: Study Plan/Top Interview 150/9 Palindrome Number/PalindromeNumber.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    9. Palindrome Number
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 52 ms (Beats 6.52% of users with C#)
    - Memory 33.04 MB (Beats 19.29% of users with C#)
*/
public class Solution
{
    public bool IsPalindrome(int x)
    {
        var original = Convert.ToString(x);
        var reversed = new string(original.ToCharArray().Reverse().ToArray());
        
        return string.Equals(original, reversed);
    }
}

File: Study Plan/Top Interview 150/0009 Palindrome Number/PalindromeNumber.java
/*  
    Problem
    --------------------------------------------------------------------------------
    9. Palindrome Number
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 7 ms (Beats 20.57% of users with Java)
    - Memory 44.34 MB (Beats 6.92% of users with Java)
*/

class Solution {
    public boolean isPalindrome(int x) {
        String original = String.valueOf(x);
        String reversed = new StringBuilder(original).reverse().toString();
        return original.equals(reversed);
    }
}
